api/src/main/java/org/openmrs/api/ConceptService.java

```diff
@@ -1970,5 +1972,47 @@ public List<ConceptSearchResult> getOrderableConcepts(String phrase, List<Locale
 	 */
 	@Authorized(PrivilegeConstants.GET_CONCEPTS)
 	boolean hasAnyConceptAttribute(ConceptAttributeType conceptAttributeType);
+
+	/**
+	 * Creates or updates the given {@link ConceptReferenceRange} in the database
+	 *
+	 * @param conceptReferenceRange ConceptReferenceRange to save
+	 * @return the created ConceptReferenceRange
+	 * 
+	 * @since 2.7.0
+	 * 
+	 * <strong>Should</strong> create a new concept reference range
+	 * <strong>Should</strong> edit an existing concept reference range
+	 */
+	@Authorized(PrivilegeConstants.MANAGE_CONCEPTS)
+	ConceptReferenceRange saveConceptReferenceRange(ConceptReferenceRange conceptReferenceRange);
+	
+	/**
+	 * This method gets Concept Reference Range by id
+	 * 
+	 * @param id conceptReferenceRange id
+	 * @return {@link ConceptReferenceRange} 
+	 * 
+	 * @since 2.7.0
+	 * 
+	 * <strong>Should</strong> get the conceptReferenceRange with the given id
+	 * <strong>Should</strong> return null if no conceptReferenceRange has the given id
+	 */
+	@Authorized(PrivilegeConstants.GET_CONCEPTS)
+	ConceptReferenceRange getConceptReferenceRangeById(Integer id);
+
+	/**
+	 * This method gets ConceptReferenceRange by concept id
+	 * 
+	 * @param conceptId conceptId 
+	 * @return list of {@link ConceptReferenceRange}
+	 *
+	 * @since 2.7.0
+	 *
+	 * <strong>Should</strong> get a list of conceptReferenceRanges with the given conceptId
+	 * <strong>Should</strong> return empty list if none of conceptReferenceRanges has the given conceptId
+	 */
+	@Authorized(PrivilegeConstants.GET_CONCEPTS)
```

Comment:
Let us get rid of this method

Reply:
This method is still in use. Did you mean :

```
ConceptReferenceRange getConceptReferenceRangeById(Integer id)
```

Reply:
You are correct. 😊